test/flow/codeGeneration.js

```diff
@@ -0,0 +1,319 @@
+          hero: ? {
+            name: string,
+          },
+        };
+      `);
+    });
+
+    it(`should generate simple query operations including input variables`, function() {
+      const context = this.compileFromSource(`
+        query HeroName($episode: Episode) {
+          hero(episode: $episode) {
+            name
+          }
+        }
+      `);
+
+      const source = generateSource(context);
+
+      expect(source).to.include(stripIndent`
+        /* @flow */
+        //  This file was automatically generated and should not be edited.
+
+        // The episodes in the Star Wars trilogy
+        export type Episode =
+          "NEWHOPE" | // Star Wars Episode IV: A New Hope, released in 1977.
+          "EMPIRE" | // Star Wars Episode V: The Empire Strikes Back, released in 1980.
+          "JEDI"; // Star Wars Episode VI: Return of the Jedi, released in 1983.
+
+
+        export type HeroNameQueryVariables = {
+          episode: ?Episode,
+        };
+
+        export type HeroNameQuery = {
+          hero: ? {
+            name: string,
+          },
+        };
+      `);
+    });
+
+    it(`should generate simple nested query operations including input variables`, function() {
+      const context = this.compileFromSource(`
+        query HeroAndFriendsNames($episode: Episode) {
+          hero(episode: $episode) {
+            name
+            friends {
+              name
+            }
+          }
+        }
+      `);
+
+      const source = generateSource(context);
+
+      expect(source).to.include(stripIndent`
+        /* @flow */
+        //  This file was automatically generated and should not be edited.
+
+        // The episodes in the Star Wars trilogy
+        export type Episode =
+          "NEWHOPE" | // Star Wars Episode IV: A New Hope, released in 1977.
+          "EMPIRE" | // Star Wars Episode V: The Empire Strikes Back, released in 1980.
+          "JEDI"; // Star Wars Episode VI: Return of the Jedi, released in 1983.
+
+
+        export type HeroAndFriendsNamesQueryVariables = {
+          episode: ?Episode,
+        };
+
+        export type HeroAndFriendsNamesQuery = {
+          hero: ? {
+            name: string,
+            friends: ?Array< {
+              name: string,
+            } >,
+          },
+        };
+      `);
+    });
+
+    it(`should generate fragmented query operations`, function() {
+      const context = this.compileFromSource(`
+        query HeroAndFriendsNames {
+          hero {
+            name
+            ...HeroFriends
+          }
+        }
+
+        fragment HeroFriends on Character {
+          friends {
+            name
+          }
+        }
+      `);
+
+      const source = generateSource(context);
+
+      expect(source).to.include(stripIndent`
+        /* @flow */
+        //  This file was automatically generated and should not be edited.
+
+        export type HeroAndFriendsNamesQuery = {
+          hero: ?(HeroFriendsFragment & {
```

Comment:
This nullable is now properly handled. `?` being a priority annotation before `&`, I had to add paren wrapping. Now the type is correct. Either it's null or either it's the union!